test/unit/resources/aws_flow_log_test.rb

```diff
@@ -0,0 +1,74 @@
+require 'helper'
+
+class EmptyAwsFlowLog < Minitest::Test
+  def setup
+    AwsFlowLog::BackendFactory.select(MockAwsFlowLog::Empty)
+  end
+
+  def test_rejects_empty_params
+    assert_raises(ArgumentError) { AwsFlowLog.new }
+  end
+
+  def test_search_hit_via_scalar_fails
+    refute AwsFlowLog.new(flow_log_id: 'fl-abcd1234').exists?
+  end
+
+  def test_rejects_unrecognized_params
+    assert_raises(ArgumentError) { AwsFlowLog.new(shoe_size: 9) }
+  end
+
+end
+
+class BasicAwsFlowLog < Minitest::Test
+  def setup
+    AwsFlowLog::BackendFactory.select(MockAwsFlowLog::Basic)
+  end
+
+  def test_search_hit
+    assert AwsFlowLog.new(flow_log_id: 'fl-abcd1234').exists?
+    assert_equal 'inspec-abcd1234', AwsFlowLog.new(flow_log_id: 'fl-abcd1234').log_group_name
+    assert_equal 'vpc-abcd1234', AwsFlowLog.new(flow_log_id: 'fl-abcd1234').resource_id
+  end
+
+  def test_search_miss
+    refute AwsFlowLog.new(flow_log_id: 'fl-12341234').exists?
+    refute AwsFlowLog.new(flow_log_id: 'fl-12341234').log_group_name
```

Comment:
Might use `assert_nil` here - IIRC, they are going to change this behavior in the next version of MiniTest. Besides, you're checking for nilly-ness, not falsiness, right?